mythril/laser/smt/__init__.py

```diff
@@ -19,14 +19,20 @@
 from mythril.laser.smt.array import K, Array, BaseArray
 from mythril.laser.smt.solver import Solver, Optimize
 
+from typing import Union, Any, Optional, List
 import z3
 
 
+Annotations = Optional[List[Any]]
+
+
 class SymbolFactory:
     """A symbol factory provides a default interface for all the components of mythril to create symbols"""
 
     @staticmethod
-    def Bool(value: bool, annotations=None):
+    def Bool(
+        value: "__builtins__.bool", annotations: Annotations = None
```

Comment:
are bool and __builtins__.bool  different? as we aren't importing bool, is there any other problem here?

Reply:
In the context of init, for some !&@# reason mypy was confusing bool with myhril.laser.smt.bool. The whole thing is very strange, as we are _not_ importing mythril.laser.smt.bool, but just importing things from it. I suppose inside __init__.py the names of .py files in the current directory are present? I am not sure. I spent half an hour trying to figure out mypy thought bool refered to mythril.laser.smt.bool, and in the end gave up and disambiguated using __builin__.bool

Reply:
For some reason mypy was confused in the context of __init__.py and thought 'bool' referes to the mythril.laser.smt.bool module, rather than the builtin type. I used __builtins__.bool here to disambiguate.